sdks/python/apache_beam/runners/worker/data_plane_test.py

```diff
@@ -99,35 +100,33 @@ def send(instruction_id, transform_id, data):
 
     # Single write.
     send('0', transform_1, b'abc')
-    self.assertEqual(
+    hc.assert_that(
```

Comment:
I don't think this change updates the test to test the correct expectations since we should be testing that multiple messages are received in order over the channel.

I believe `send` doesn't have the correct semantics since we want to have a persistent stream for the life of this test and to control when it gets closed after sending for a specific `instruction,transform` pair is done.

Reply:
Also, we may want to have a sleep in the time based test to force a flush sometimes between messages (which likely means we should test with more messages).

Reply:
So is the correct fix to create `stream` outside of `send` and re-use it? Or do we need a stream per assertion?

Reply:
Yes, the stream should be created outside of send so that its lifetime can be explicitly managed in the test based upon what the test is trying to do.

Reply:
I implemented some of your suggestions Luke. PTAL